### PR TITLE
[BUGFIX] Product of Pauli Operators was defined wrong.

### PR DESCRIPTION
**Pull request by isolatedinformation**
- Corrected the output of the prodcuts
- line 254 in circuit.py should be multiplication not division.
- made relevant changes to rotation.py tests

### CODE REVIEW

File: src/lsqecc/pauli_rotations/circuit.py [hunk 1]
```diff
@@ -224,7 +224,7 @@ def commute_pi_over_four_rotation(self, index: int) -> None:
 
             # Flip the phase if product of coefficients is negative
             # Product of coefficients will always be either i or -i (see issues #28 for proof)
-            product_of_coefficients /= 1j
+            product_of_coefficients *= 1j
```
Comment (alexnguyenn) [inline src/lsqecc/pauli_rotations/circuit.py:227]: I am not 100% sure rn but I think the reason we use `*= 1j` is for checking whether we are getting i or -i.

File: src/lsqecc/pauli_rotations/rotation.py [hunk 2]
```diff
@@ -78,10 +78,10 @@ def multiply_operators(a: "PauliOperator", b: "PauliOperator"):
 ] = {
     (PauliOperator.Z, PauliOperator.X): (1j, PauliOperator.Y),
     (PauliOperator.X, PauliOperator.Z): (-1j, PauliOperator.Y),
-    (PauliOperator.Z, PauliOperator.Y): (1j, PauliOperator.X),
-    (PauliOperator.Y, PauliOperator.Z): (-1j, PauliOperator.X),
-    (PauliOperator.Y, PauliOperator.X): (1j, PauliOperator.Z),
-    (PauliOperator.X, PauliOperator.Y): (-1j, PauliOperator.Z),
+    (PauliOperator.Y, PauliOperator.Z): (1j, PauliOperator.X),
```
Comment (alexnguyenn) [inline src/lsqecc/pauli_rotations/rotation.py:81]: how did I even miss this… 😱

**Review by gwwatkin — APPROVED** [review]
Works as agreed over zoom

File: src/lsqecc/pauli_rotations/rotation.py (hunk 2 above)
Comment (gwwatkin) [reply src/lsqecc/pauli_rotations/rotation.py:81]: I think there was some confusion about what this table is, we need to comment with what is the invariant between the tuple elements:
`(A,B),(k,C) => A*B = k*C`

**Pull request merged**